pyro/infer/mcmc/trace_kernel.py

```diff
@@ -0,0 +1,25 @@
+from __future__ import absolute_import, division, print_function
+
+from abc import abstractmethod, ABCMeta
+
+from six import add_metaclass
+
+
+@add_metaclass(ABCMeta)
+class TraceKernel(object):
```

Comment:
Do you think `TraceKernel` should be a `Distribution`?

Reply:
Hmm..that's an interesting thought. The semantics are somewhat different in that we cannot generate a sample directly, since it needs an existing sample.